resolve/index.go

```diff
@@ -110,22 +110,22 @@ type RuleIndex struct {
 
 // ruleRecord contains information about a rule relevant to import indexing.
 type ruleRecord struct {
-	kind  string
-	label label.Label
+	Kind  string      `json:"kind"`
```

Comment:
Since you probably know this part of the code better than me by now: Is it correct that we don't expose any fields or methods that return a `ruleRecord`? Otherwise making the fields public could end up adding new public API.

Reply:
AFAIU `ruleRecord` is private to `RuleIndex` in this file. It is used to persist the data collected from languages and never directly exposed.

I think `FindRulesByImport` is a good example which searches the data (actually searches `importMap` which is computed from the `ruleRecord`s), then returns a completely different `FindResult` struct and never exposes `ruleRecord`s or `importMap`...

So `ruleRecord` is private, but it's fields are public so the json encoding works 🤷

Reply:
Sounds good, thanks for confirming!